2-Draft-Accepted/RFC00NN-PowerShellCore-Installer-Module.md

```diff
@@ -0,0 +1,100 @@
+## Motivation
+
+Installation and configuration of PowerShell Core on Windows is tricky due to the wide range of Windows versions that we support (7-10 and Server 2008R2-2016)
+and because of the usage of some "system" concepts shared by Windows PowerShell like the WinRM remoting plugin.
+
+## Goals
+
+* I can install PowerShell Core across all supported versions of Windows and Windows PowerShell *at scale*.
+* I can install PowerShell Core on certain versions of Windows and Windows PowerShell without remembering highly arcane commands or manually downloading a package from a well-known location.
+* I can update PowerShell Core in-place after it's already been installed.
+* I can uninstall PowerShell Core using Windows PowerShell
+* I can register PowerShell Core as a WinRM remoting endpoint across all supported versions of Windows and Windows PowerShell *at scale*.
+
+## Specification
+
+Ship a module on Gallery called `PowerShellCoreProvider` which contains the following cmdlets:
+
+* Install-PowerShellCore
+  * Download and install the latest PowerShell Core MSI and install it using the MSI PackageManagement provider
+* Update-PowerShellCore
+  * If PowerShell Core is already installed, this will grab the latest PowerShell Core MSI and do an in-place upgrade of the MSI-installed instance on the machine.
+* Uninstall-PowerShellCore
+  * Uninstall the PowerShell Core MSI using the MSI PackageManagement provider.
+* Register-PowerShellCoreEndpoint
+  * Call the Install-PowerShellRemoting.ps1 script in PowerShell Core's `$PSHome`.
+
+These cmdlets allow me to do things like:
+
+```powershell
+Invoke-Command -ComputerName $TargetNode -Credential DOMAIN\BestAdmin -ScriptBlock {Install-Module PowerShellCore -Force; Install-PowerShellCore; Register-PowerShellCoreEndpoint}
+```
+
+`PowerShellCoreProvider` should start as a 0.1 release so that we can make breaking changes as we gather feedback in PowerShell Core 6.0 beta.
+It should also ship as a standalone module outside of the PowerShell Core package.
+
+Right now, I think it's perfectly okay to ship all of these with nothing but common parameters.
+Based on feedback to this RFC, we should prioritize a set of parameters that might be useful.
+(For example, we may eventually add a `-Source` to `Install/Upgrade-PowerShellCore` that points to `GitHub` and some other, more official, source.)
+
+## Alternate Proposals and Considerations
+
+### Tracking Side-by-Side Installations of Powershell Core
+
+In the future, it may become necessary for this provider to help manage the installation of multiple copies of PowerShell Core (or even Windows PowerShell).
+Similar to `venv` for Python or `rbenv` for Ruby,
+this feature might enable users to call into various versions of PowerShell, or
+handle symlinking of the "primary" PowerShell to a specific version.
+
+However, until PowerShell Core has moved to a new version (e.g. 7.x or 6.1),
+this feature isn't necessary and is therefore out of scope for this RFC.
```

Comment:
You can always have side-by-side betas (or any version really) by unzipping and putting into a different folder, however, non-developers are unlikely to do this and any installer is targeting the mainstream.